SecretManagement.LastPass.Extension/SecretManagement.LastPass.Extension.psm1

```diff
@@ -195,3 +289,59 @@ function Test-SecretVault
     $status = Invoke-lpass 'status' -ErrorAction SilentlyContinue
     return ($status -match "Logged in as .*")
 }
+
+function Get-SimpleSecret {
+    [CmdletBinding()]
+    param (
+        $Fields,
+        $Note
+    )
+    
+    $IsCredentials = $Fields.where( { $_.key -in @('Username', 'Password') }, 'first').count -eq 1
+    if ($IsCredentials) {
+        $username = $Fields.Where( { $_.key -eq 'Username' }, 'first') | Select-Object -ExpandProperty value
+        if ($null -eq $username) { $username = '' }
+        $password = $Fields.Where( { $_.key -eq 'Password' }, 'first') | Select-Object -ExpandProperty value
+        if ($null -eq $password) { $password = '' }
+        if ("" -ne $password) { $password = $password | ConvertTo-SecureString -AsPlainText -Force }
+        $output = [System.Management.Automation.PSCredential]::new($username, $password)
+    }
+    else {
+        if ($null -eq $Note) { $output = "" } else { $output = $Note }
+    }
+    return $output
+}
+
+function Get-ComplexSecret {
+    [CmdletBinding()]
+    param (
+        $Fields,
+        $Note
+    )
+    $Dupes = ($Fields | Group-Object key).Where( { $_.Count -gt 1 })
+    
+    if ($Dupes.count -gt 0) {
+        Write-Verbose 'Creating case-sensitve hashtable'
+        $Output = [hashtable]::new([System.StringComparer]::InvariantCulture)
+    } else {
+        $Output = @{}
+    }
+    
+    if (![String]::IsNullOrEmpty($Note)) { 
+        $Output.Notes = $Note
+        $Fields = $Fields | Select-Object -SkipLast 1
+    }
+
+    Foreach ($f in $Fields) {
+        try {
+            $Output.Add($f.key, $f.value) 
+        }
+        catch {
+            Write-Warning "$($f.key) field was not added."
+        }
+    }
+    if ($null -ne $Output.NoteType -and $DefaultNoteTypeMap.ContainsKey($Output.NoteType)) {
+        $Output.NoteType = $DefaultNoteTypeMap.Item($Output.NoteType)
+    }
+    return $Output
+}
```

Comment:
```suggestion
}

```
nit: newline at EOF

Reply:
TIL:  I mostly work on Windows operating system. 
Good to know and done.

Reply:
I just don't like seeing the icon on GitHub when there isn't a newline :)